serverless/components
# Make Components Removal Idempotent

This PR adds idempotency to the remove function of the following components:

- aws-s3-bucket
- s3-policy
- aws-iam-policy
- aws-dynamodb
- aws-apigateway

## Testing

1. deploy any of those components
2. copy the state
3. run remove the component
4. after the remove is complete replace the state file with the original copy.
5. run remove again.
6. The removal should not throw an error and should complete successfully.
7. The state file should be updated to reflect the component has been removed.
